### PR TITLE
Fix cmake installs since we added clog.

### PR DESCRIPTION
**Pull request by ThadHouse**
This uses a tracepoint provider for lttng, which is a common practice, coreclr does the same thing.

### CODE REVIEW

File: scripts/package-distribution.ps1 [hunk 1]
```diff
@@ -82,8 +82,7 @@ foreach ($Build in $AllBuilds) {
     } else {
         $Binaries += Join-Path $ArtifactsDir "libmsquic.so"
         #Temporary until we fix CLOG
-        $Binaries += Join-Path $ArtifactsDir "libPLATFORM_CLOG_LIB.so"
-        $Binaries += Join-Path $ArtifactsDir "libCORE_CLOG_LIB.so"
+        $Binaries += Join-Path $ArtifactsDir "libmsquictraceprovider.so"
```
Comment (nibanks) [inline scripts/package-distribution.ps1:85]: Can this be named `libmsquic.traceprovider.so` or (since I think it's only necessary when we use lttng) `libmsquic.lttng.so`?

File: src/platform/platform_linux.c [hunk 2]
```diff
@@ -63,6 +66,60 @@ QuicPlatformSystemLoad(
...
+    // https://github.com/dotnet/coreclr/blob/ed5dc831b09a0bfed76ddad684008bebc86ab2f0/src/pal/src/misc/tracepointprovider.cpp#L106
+    //
+
+    int fShouldLoad = 1;
+    // Check if loading the LTTng providers should be disabled.
+    char *disableValue = getenv("QUIC_LTTng");
+    if (disableValue != NULL)
+    {
+        fShouldLoad = strtol(disableValue, NULL, 10);
+    }
```
Comment (nibanks) [inline src/platform/platform_linux.c:80]:
```suggestion
    if (disableValue != NULL) {
        fShouldLoad = strtol(disableValue, NULL, 10);
    }
```

File: src/platform/platform_linux.c [hunk 3]
```diff
@@ -63,6 +66,60 @@ QuicPlatformSystemLoad(
     void
     )
 {
+    //
+    // Following code is modified from coreclr.
+    // https://github.com/dotnet/coreclr/blob/ed5dc831b09a0bfed76ddad684008bebc86ab2f0/src/pal/src/misc/tracepointprovider.cpp#L106
+    //
+
+    int fShouldLoad = 1;
```
Comment (nibanks) [inline src/platform/platform_linux.c:74]: None of these variables follow our style.

File: scripts/package-distribution.ps1 (hunk 1 above)
Comment (ThadHouse) [reply scripts/package-distribution.ps1:85]: Yeah I'll do that.

File: scripts/prepare-machine.ps1 [hunk 4]
```diff
@@ -21,7 +21,7 @@ param (
     [string]$Configuration
 )
 
-#Requires -RunAsAdministrator
+##Requires -RunAsAdministrator
```
Comment (nibanks) [inline scripts/prepare-machine.ps1:24]: Was this change on purpose?

Comment (ThadHouse) [reply scripts/prepare-machine.ps1]: Nope. Fixed.

File: src/bin/CMakeLists.txt [hunk 5]
```diff
@@ -34,9 +34,19 @@ set(PUBLIC_HEADERS
...
-#install(FILES ${CMAKE_BINARY_DIR}/msquic-config.cmake DESTINATION ${msquic_dest})
-#install(EXPORT msquic DESTINATION ${msquic_dest})
+configure_file(msquic-config.cmake.in ${CMAKE_BINARY_DIR}/msquic-config.cmake)
+
+install(FILES ${CMAKE_BINARY_DIR}/msquic-config.cmake DESTINATION ${msquic_dest})
+
+install(EXPORT msquic DESTINATION ${msquic_dest})
+
+if(WIN32)
+    add_library(msquic.lttng INTERFACE)
```
Comment (nibanks) [inline src/bin/CMakeLists.txt:47]: Why are you doing this on Windows? There is no lttng on Windows.

File: src/platform/platform_linux.c [hunk 6]
```diff
@@ -63,6 +66,61 @@ QuicPlatformSystemLoad(
     void
     )
 {
+    //
+    // Following code is modified from coreclr.
+    // https://github.com/dotnet/coreclr/blob/ed5dc831b09a0bfed76ddad684008bebc86ab2f0/src/pal/src/misc/tracepointprovider.cpp#L106
+    //
+
+    int ShouldLoad = 1;
+    // Check if loading the LTTng providers should be disabled.
```
Comment (nibanks) [inline src/platform/platform_linux.c:75]: Comments in this function don't follow our style.

File: src/bin/CMakeLists.txt (hunk 5 above)
Comment (ThadHouse) [reply src/bin/CMakeLists.txt:47]: I need a target on windows, since downstream executables use this target to make loading slightly easier. This just creates an empty target, so every downstream use doesn't require an if check.

Comment (nibanks) [reply src/bin/CMakeLists.txt:47]: Ok. Though, I expect this will eventually move into clog somehow...
